crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_numeric_union.rs

```diff
@@ -37,6 +44,13 @@ use crate::checkers::ast::Checker;
 /// def foo(x: float | str) -> None: ...
 /// ```
 ///
+/// ## Fix safety
+/// This rule's fix is marked as safe for most cases; however, the fix will
+/// flatten nested unions type expressions into a single top-level union.
```

Comment:
Reading the documentation, I get the impression the fix isn't safe because it flattens nested union expressions. But I understand that this doesn't make the fix unsafe, it's just an extension where the fix goes beyond what one might expect.

Reply:
Exactly. I've reused the phrasing that Charlie used for another rule.

The nesting is a feature to allow cases such as these:
```python
ReadOnlyMode         = Literal["r", "r+"]
WriteAndTruncateMode = Literal["w", "w+", "wt", "w+t"]
WriteNoTruncateMode  = Literal["r+", "r+t"]
AppendMode           = Literal["a", "a+", "at", "a+t"]

AllModes = Literal[ReadOnlyMode, WriteAndTruncateMode,
                   WriteNoTruncateMode, AppendMode]
```

As a consequence this redundant nesting is also allowed, but can be safely fixed:

```python
AllModes = Literal[Literal["r", "r+"], Literal["w", "w+", "wt", "w+t"],
                   Literal["r+", "r+t"], Literal["a", "a+", "at", "a+t"]]
```

See https://typing.readthedocs.io/en/latest/spec/literal.html#legal-and-illegal-parameterizations

We probably should have a rule for this (for both `union` and `literal`).

Reply:
I tweaked the comment a bit on all three rules.